lib/cucumber/core/event_bus.rb

```diff
@@ -0,0 +1,61 @@
+require 'cucumber/core/events'
+
+module Cucumber
+  module Core
+
+    # Event Bus
+    #
+    # Implements an in-process pub-sub event broadcaster allowing multiple observers
+    # to subscribe to events that fire as your tests are executed.
+    #
+    class EventBus
+      attr_reader :event_types
+
+      # @param [Hash{Symbol => Class}] a hash of event types to use on the bus
+      def initialize(registry = Events.registry)
+        @event_types = registry.freeze
+        @handlers = {}
+      end
+
+      # Register for an event. The handler proc will be called back with each of the attributes
+      # of the event.
+      def on(event_id, handler_object = nil, &handler_proc)
```

Comment:
Fixed in https://github.com/cucumber/cucumber-ruby-core/pull/106/commits/8d19559078ea6fd1ff50d73a84febad13253a83b.